### PR TITLE
Automate the release process: Enforce prettyness of source; auto-publish docs

### PR DESCRIPTION
**Pull request by nfriedly**
This commit is part of https://github.com/jlongster/prettier/issues/581:

* An extra check is now run during the build to ensure that the source code of prettier meets it's own standards. If not, the build is failed. This isn't quite what was requested (automatically send a PR), but it's a good first step if nothing else.

* After a successful build, automatically generate and publish the docs.

The format check works as is, but the auto-publishing requires a bit of extra work from someone with commit access to this repo: 
1. Create a token at https://github.com/settings/tokens with the `public_repo` scope
2. Remove the existing `env.global.secure` in `.travis.yml`
3. run `travis encrypt --add -r jlongster/prettier 'GH_TOKEN=xxxxxxxxxxxxxxxxxxx'` with your new token from step 1.

(Or I can do it if you want to grant me commit access.)

Known issues:
   * The build is already failing due to missing lines in the expected output - https://travis-ci.org/jlongster/prettier/jobs/199263112#L458
   * The format check adds another failure because the source does not currently live up to it's own standards.  This should be easy to fix, just run `npm run format:all` and commit the changes.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
   - node_modules
-script: npm test -- --runInBand
+script:
+ - npm test -- --runInBand
+ - scripts/check_format.sh
+after_success:
+- scripts/publish_docs.sh
+env:
+  global:
+    secure: dIC0jXp5WzC7dRvjggGFLP8N4BB6H3nx2txfJ53w/V912LFBiWFNeMfJ5OB5AbyMXMr+2zEADrWO8qEPObPfvCQDcOIEfwqZyiqbRz99aewejJAlL2oPgbFv85eyU5N1iPj1WxRbMHak/mAN9iegcJcbkaKXzXrrlXua/qbbOiETtXRprL4p0KyngcZkg3zwPOrLsyd4lLBiiy074hqqfuzdnYQgo4Rovqa6BaEKizm3bbhGUoiYhDahH3Y19i9NitqlpV3gnckE29zdZYSWtBQh5AqjRvCg59GR8Gs9rCxea3fJb0l22pEGDuDGb3n9QxGSNrKAgxyTvgtLZnIxWyaVaajwjQbAI76VsP0uSqjKgjZUP4sU5l8amnaKx74W7dfn7J3+vpSctkA75SZpnXYFDZ+LBp8WNe1kMWeDp64KmLzoGX/8HHsoc+Adljywj2zHuADAr+wAxObW+PVq8jJ7/fUSsGZy4qjjHxxzIZ2u2jrraPFMrcl4BVhSSDI8UpWTvzjITuEzsYMpcy7V4SgfvePrzkVa4w9a2tcPfvhDgOge6YwDqD//MV0lcuOjXUcjFQlmcNn3BZd01bwS3/JVt9IVOnrZ7jV9l9CqsZoQpsa4T2dTtm8kvBmN1Mf0PvGvrC+cF8q/DxFi4AjFmcgRaWDpPzoW9czkNEE+Swk=
```
Comment (nfriedly) [inline .travis.yml]: You'll want to delete this and replace it with your own, but I left mine there to show an example of what it should look like.

**Comment by Daniel15** [comment]
Wouldn't it be cleaner to do this on a `gh-pages` branch? It's really icky to commit generated code to master and makes the Git history pretty messy. It's still pretty bad to do it in a branch, but I totally understand that it's due to limitations with GitHub Pages. At least if it's on a gh-pages branch, it's mostly out-of-sight from people working on Prettier itself.

**Comment by nfriedly** [comment]
Yea, I usually do things on `gh-pages`, but I don't have access to set it up here, so I just went with the existing structure. 

Anyone with commit rights can move things over to gh-pages, and I can help update the scripts then.

**Comment by Daniel15** [comment]
I'd strongly suggest getting one of the maintainers to move the site to a gh-pages branch as you'll have a much better experience doing that instead of automatically committing to master. Just my opinion 👍

**Comment by vjeux** [comment]
Hey, I'm soooo sorry I didn't review this at all. I feel super guilty about it :(

The idea was to automate the process where we were doing a release almost every day, but now that we are getting a lot of adoption, we actually want to dramatically slow down the number of releases and make them more high quality with a proper blog post. So doing the release isn't that big an overhead and it's better to keep it manual.

I'm sorry about it :x

**Comment by nfriedly** [comment]
heh, I had wondered about this once or twice. No worries though; congrats on shipping 1.0 and good luck with the project!

**Pull request closed without merging**